src/tools/concurrency/TracingActors.java

```diff
@@ -44,25 +44,33 @@
 
     public TracingActor(final VM vm) {
       super(vm);
-      this.actorId = IdGen.getAndIncrement();
+      this.activityId = TracingActivityThread.newEntityId();
+      assert this.activityId >= 0;
       if (VmSettings.SNAPSHOTS_ENABLED) {
         snapshotRecord = new SnapshotRecord();
       }
     }
 
-    protected TracingActor(final VM vm, final int id) {
+    protected TracingActor(final VM vm, final long id) {
       super(vm);
-      this.actorId = id;
+      this.activityId = id;
     }
 
     public final int getActorId() {
```

Comment:
Could we just remove this method?

Reply:
My idea was to leave it in so that I don't have to touch the snapshotting code (making rebase of snapshotting easier).

Reply:
Ah, right, you mentioned that. Though, could you add a comment with a todo?